e2e_tests/test_suites/inventoryreporting/inventory_reporting.go

```diff
@@ -0,0 +1,152 @@
+//  Copyright 2018 Google Inc. All Rights Reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package inventoryreporting
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"path"
+	"regexp"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/GoogleCloudPlatform/compute-image-tools/go/e2e_test_utils/junitxml"
+	"github.com/GoogleCloudPlatform/osconfig/e2e_tests/compute"
+	"github.com/GoogleCloudPlatform/osconfig/e2e_tests/config"
+	gcpclients "github.com/GoogleCloudPlatform/osconfig/e2e_tests/gcp_clients"
+	testconfig "github.com/GoogleCloudPlatform/osconfig/e2e_tests/test_config"
+	"github.com/GoogleCloudPlatform/osconfig/e2e_tests/utils"
+	api "google.golang.org/api/compute/v1"
+)
+
+const (
+	testSuiteName = "OSInventoryReporting"
+)
+
+// TestSuite is an OSInventoryReporting test suite.
+func TestSuite(ctx context.Context, tswg *sync.WaitGroup, testSuites chan *junitxml.TestSuite, logger *log.Logger, testSuiteRegex, testCaseRegex *regexp.Regexp) {
+	defer tswg.Done()
+
+	if testSuiteRegex != nil && !testSuiteRegex.MatchString(testSuiteName) {
+		return
+	}
+
+	testSuite := junitxml.NewTestSuite(testSuiteName)
+	defer testSuite.Finish(testSuites)
+
+	logger.Printf("Running TestSuite %q", testSuite.Name)
+
+	var wg sync.WaitGroup
+	tests := make(chan *junitxml.TestCase)
+	for _, setup := range headImageTestSetup() {
+		wg.Add(1)
+		go inventoryReportingTestCase(ctx, setup, tests, &wg, logger, testCaseRegex)
+	}
+
+	go func() {
+		wg.Wait()
+		close(tests)
+	}()
+
+	for ret := range tests {
+		testSuite.TestCase = append(testSuite.TestCase, ret)
+	}
+
+	logger.Printf("Finished TestSuite %q", testSuite.Name)
+}
+
+func runInventoryReportingTest(ctx context.Context, testSetup *inventoryTestSetup, testCase *junitxml.TestCase, logwg *sync.WaitGroup) {
+	testCase.Logf("Creating compute client")
+
+	computeClient, err := gcpclients.GetComputeClient()
+	if err != nil {
+		testCase.WriteFailure("Error getting compute client: %v", err)
+		return
+	}
+
+	testSetup.hostname = fmt.Sprintf("inventory-test-%s-%s", path.Base(testSetup.testName), utils.RandString(5))
+
+	var metadataItems []*api.MetadataItems
+	metadataItems = append(metadataItems, testSetup.startup)
+	metadataItems = append(metadataItems, compute.BuildInstanceMetadataItem("enable-os-inventory", "true"))
+	metadataItems = append(metadataItems, compute.BuildInstanceMetadataItem("osconfig-disabled-features", "tasks,guestpolicies"))
```

Comment:
You should only need enable-os-inventory, or you can just enable all of osconfig and then specifically disable tasks and guest policies if you want

Reply:
In practice, without `osconfig-disabled-features` the agent on the test VMs would attempt to listen for patch task and run the guest policies task (which could take some time). I'll share some screenshots with you offline.

Reply:
We must have it enabled at the project then, not that big an issue.